Refactor/sign process pages

geon-2 opened the pull request:
## ✨ Issue Number

- close #16 

## 🗂️ Details

### 회원 프로세스
와이어프레임 상의 회원가입 프로세스가 기술적 문제로 변경된 부분에 적합하지 않아 UX를 고려한 적합한 프로세스가 되도록 변경하였음
- 서버 상의 이메일, 비밀번호 인증이 보조출연자, 업체 상관 없이 통합됨
- 타투 저장 방식이 불필요한 서버 자원을 사용한다고 판단됨

- [x] LoginPage
- [x] MainFormPage: 이름, 이메일, 비밀번호, 비밀번호 확인
- [x] AuthenticationPage: 본인인증
- [x] SelectUserTypePage
- [x] MemberInfoFormPage: 보조출연자 정보
- [x] CompanyInfoFormPage: 소속 업체 선택
- [x] AccountFormPage: 계좌 정보
- [x] TattooFormPage, TattooSelectFormPage: 타투 여부, 타투 부위
- [x] TermModal: 약관 동의 모달

프로세스
1. LoginPage
2. 회원가입 클릭
3. MainFormPage
4. AuthenticationPage
5. SelectUserTypePage
6. 1. 보조출연자 선택
6. 1. 1. MemberInfoFormPage
6. 1. 2. TattooFormPage
6. 1. 3. (타투 있음 선택 시) TattooSelectFormPage
6. 1. 4.  AccountFormPage
6. 2. 업체 선택
6. 2. 1. CompanyInfoFormPage
7. TermModal
8. 메인으로

### 추가 항목
UI 구현에 필요한 컴포넌트 추가

atoms
- Window: PWA 화면 프레임
- Container: flexbox
- Margin: 여백

mocules
- TypeSelectButton: 빅 사이즈 선택버튼
- DropDownInput: 모바일 환경에 맞춘 드롭다운 선택창 (Cupertino DropDownPicker Design)
- SIgnWindow: 회원가입 화면 프레임

## ✅ Need Review

- [ ] 회원 프로세스가 사용자 측면에서 적합한가
- [ ] 추가된 atoms 컴포넌트들이 재사용하기에 적합한가
- [ ] 기존에 요소들과 중복되는 부분은 없는가

## 📸 Screenshot


## 📚 Reference

geon-2 committed:
Refactor: Input, DropDownInput 컴포넌트 추가
회원 관련 프로세스 구현을 위한 컴포넌트 추가
- Input: 텍스트 입력 input field
- DropDownInput: cupertino design DropDownPicker

geon-2 committed:
Chore: Install react-hook-form, react-icons
react libraray 설치
- react-hook-form: for mananging formData
- react-icons: for including icons

geon-2 committed:
Refactor: Add new atoms
로그인 프로세스 구현을 위한 atom 요소 추가
- Container, Window
- HistoryBackButton
- Margin
- Form
    - Input
    - InputField

geon-2 committed:
Refactor: Add DropDownInput
cupertino 디자인 드롭다운 input 구현

geon-2 committed:
Refactor: Add color.ts
테마 관리를 위한 color 정의

geon-2 committed:
Refactor: Add login page
로그인 페이지 구현

geon-2 committed:
Refactor: Add utill for checking platform
반응형을 위한 플랫폼 확인 유틸함수 추가

geon-2 committed:
Refactor: Add SignWindow components
회원가입 프로세스 공통 화면 구현
- 회원가입 header
- submit button

geon-2 committed:
Refactor: Add MainFormPage
회원 기본 정보 입력 페이지 추가

geon-2 committed:
Refactor: Add AuthenticationPage
전화번호 인증 페이지 추가

geon-2 committed:
Refactor: Add SelectUserTypePage
유저 유형 선택 페이지 추가
- TypeSelectButton: 큰 사이즈 select button

geon-2 committed:
Refactor: Add MemberInfoFormPage
보조출연자 정보 입력 페이지 추가

geon-2 committed:
Refactor: Add TattooFormPage, TattooSelectFormPage
타투 선택 페이지 추가
- TattooFormPage: 타투 여부 선택
- TattooSelectFormPage: 타투 부위 선택
\# 16

geon-2 committed:
Refactor: Add AccountFormPage
계좌 인증 페이지 추가
\# 16

geon-2 committed:
Refactor: Add CompanyInfoFormPage
회사 선택 페이지 추가

geon-2 committed:
Refactor: Add TermModal, routing
약관 동의 모달 구현 및 회원 프로세스 관련 라우팅 추가
\# 16

geon-2 committed:
Refactor: Update ThemePreview
react-hook-form 설치로 수정된 InputField 사용법 추가

geon-2 committed:
Refactor: Add CompanyShootManageDetailPage
업체 촬영 관리 상세 메뉴 페이지 추가
\# 18

geon-2 committed:
Merge branch 'develop' into refactor/manage-pages

geon-2 committed:
Refactor: Add ActorListPage
업체 촬영 관리 출연자 목록 페이지 추가
- ScrollingList: 스크롤 가능한 리스트
- Filter: 출연자 목록 검색 필터
- SearchInputField: 검색 input field

geon-2 committed:
Refactor: Add ClothesConfirmStatusListPage
업체 현장 관리 의상 컨펌 페이지 추가

geon-2 committed:
Chore: Install vconsole
모바일 환경 디버깅을 위한 로그 출력 패키지 추가

geon-2 committed:
Chore: Set https options
모바일 카메라 테스트를 위한 https 호스팅 허용 옵션 추가

geon-2 committed:
Refactor: Add CameraPage
의상 컨펌 및 QR 인증을 위한 카메라 페이지 추가

geon-2 committed:
Refactor: Update CameraPage
뒤로가기, 카메라 전/후면 변경 기능 구현

geon-2 committed:
Refactor: Add the feature of scanning qr-code
QR 코드 스캔 기능 추가
- MemberShootManageDetailPage: 보조출연자 촬영 관리 상세 페이지 생성 및 QR 추가
- AttendancePage: 출석 인증 페이지에 QR 스캐너 추가

goodbyebada commented on src/index.css:
```diff
@@ -51,3 +51,15 @@ button {
   border: solid 1px white;
 }
+
+/* Chrome, Safari, Edge, Opera */
+input[type="number"]::-webkit-inner-spin-button,
+input[type="number"]::-webkit-outer-spin-button {
+  -webkit-appearance: none;
+  margin: 0;
+}
+
+/* Firefox */
+input[type="number"] {
+  -moz-appearance: textfield;
+}
```
크로스 브라우저 호환성 고려하셨네요 👍

geon-2 committed:
Refactor: Update theme files
- constants/Color.ts 파일 styled/color.ts로 통합
- Window, Container 파일에서 분리
- SignWindow, MainWindow 추가
- MainWindow에 BottomNavigation 추가

geon-2 committed:
Refactor: Update ClothesConfirmStatusListPage
검색 기능 추가

geon-2 committed:
Merge branch 'develop' into refactor/sign-process-pages

geon-2 committed:
Merge branch 'refactor/manage-pages' into refactor/sign-process-pages

geon-2 committed:
Refactor: Add kakao login, profile and setting pages
카카오 로그인 기능
- react-kakao-login 라이브러리로 구현

마이페이지, 설정 페이지 구현

goodbyebada commented on src/pages/Sign/MemberInfoFormPage.tsx:
```diff
@@ -0,0 +1,95 @@
+        name="weight"
+        control={control}
+        placeholder="몸무게"
+        rules={{
```
몸무게가 세자리인 분들은 가입할 수 없네요 ㅜㅜ 세자리 수까지 입력 가능하게 수정이 필요해보입니다!

goodbyebada commented on src/pages/Sign/LoginPage.tsx:
```diff
@@ -0,0 +1,162 @@
+          }}
+        />
+        <Margin size={15} />
+        <InputField
+          name="password"
```
```suggestion
          type="password"
          name="password"
```

추후에 type 설정 추가 부탁드려요!

goodbyebada commented on src/components/atoms/Container.tsx:
```diff
@@ -0,0 +1,94 @@
+    );
+  },
+);
+Container.displayName = "Container";
```
displayName 속성을 정의하신 이유가 따로 있을까요?

geon-2 replied:
typescript에서 컴포넌트에 ref를 넘겨줄 때 필요해서 넣었어요. forwardRef의 파리미터에 익명 함수로 넘겨줄 때는 컴포넌트 명이 정의되어 있지 않아서, 따로 명시해주어야 해당 이름을 모듈로써 사용할 수 있더라구요.

goodbyebada commented on src/components/atoms/Container.tsx:
```diff
@@ -0,0 +1,94 @@
+import React, { forwardRef, HTMLAttributes } from "react";
+import styled from "styled-components";
+
+interface ContainerProps extends HTMLAttributes<HTMLDivElement> {
+  children: React.ReactNode;
+  background?: string;
+  paddingVertical?: number;
+  paddingHorizontal?: number;
+  flex?: number;
+  flexDirection?: "row" | "column";
+  justifyContent?:
+    | "flex-start"
+    | "flex-end"
+    | "center"
+    | "space-between"
+    | "space-around";
+  alignItems?:
+    | "flex-start"
+    | "flex-end"
+    | "center"
+    | "space-between"
+    | "space-around";
+  flexWrap?: boolean;
+}
```
다양한 레이아웃을 지원할 수 있어 좋습니다

goodbyebada commented on src/components/atoms/Container.tsx:
```diff
@@ -0,0 +1,94 @@
+    }: ContainerProps,
+    ref,
+  ) => {
+    return (
+      <StyledContainer
+        background={background}
+        ref={ref}
+        flex={flex}
+        paddingHorizontal={paddingHorizontal}
+        paddingVertical={paddingVertical}
+        justifyContent={justifyContent}
+        alignItems={alignItems}
+        flexDirection={flexDirection}
+        flexWrap={flexWrap}
+        {...props}
+      >
```
현재 paddingHorizontal과 같은 커스텀 props가 DOM에 요소에 전달되면서 콘솔창에 에러메시지가 발생하고 있습니다. 후에 개발할때 디버깅하기가 어려울 것 같아요.
소문자를 사용하거나 $ prefix를 추가해 수정해야할 것 같습니다!

goodbyebada commented on src/components/mocules/manage/CardItem.tsx:
```diff
@@ -0,0 +1,116 @@
+
+const ClothesStatus = styled.div<CardStatus>`
+  background: ${({ status }) =>
+    status ? "#35EB17" : BACKGROUND_COLORS.disabled};
```
"#35EB17" 도  BACKGROUND_COLORS.disabled처럼 상수화하면 통일성 있을 것 같습니다

goodbyebada commented on src/components/mocules/MainWindow.tsx:
```diff
@@ -0,0 +1,335 @@
+const ICON_SIZE = 40;
+const ACTIVE_COLOR = COMMON_COLORS.main;
+const INACTIVE_COLOR = FONT_COLORS.gray;
+const headerHeight = 155;
```
임시로 하신 걸까요? 헤더 UI가 너무 두꺼워서 60~70으로 수정해도 될 것 같아요!

goodbyebada commented on src/components/mocules/manage/SearchInputField.tsx:
```diff
@@ -0,0 +1,39 @@
+const SearchInputField = ({ value, setValue }: SearchInputFieldProps) => {
+  return (
+    <InputWrapper>
+      <Input value={value} onChange={(e) => setValue(e.target.value)} />
```
debounce를 사용하면 불필요한 렌더링을 줄일 수 있을 것 같습니다

goodbyebada commented on src/components/mocules/manage/Filter.tsx:
```diff
@@ -0,0 +1,117 @@
+
+const Filter = ({ items, setValue }: FilterProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [currentIndex, setCurrentIndex] = useState(-1);
+
```
type을 number로 명시하고 === 으로 비교하면 더 안전할 것 같아요

goodbyebada commented on src/components/mocules/SignWindow.tsx:
```diff
@@ -0,0 +1,48 @@
+  title: string;
+  subTitle?: string;
+  submitText?: string;
+  buttonProps: Parameters<typeof MainButton>[0];
```
MainButton의 props를 interface ButtonProps로 import해서 적용하는 방법도 있는데, 이 방법을 선택하신 이유가 궁금합니다!

geon-2 replied:
동적 추출을 위해서 저렇게 구현했는데, 지금보니 큰 의미가 보이네요. 해당 컴포넌트 내에서 사용될 parameter만 불러오려고 직접 타입을 추출하는 방식으로 구현했었어요.

goodbyebada commented on src/pages/Sign/AuthenticationPage.tsx:
```diff
@@ -0,0 +1,174 @@
+          </Container>
+        ) : (
+          <Container>
+            <InputField
+              name="phone"
+              control={control}
+              placeholder="전화번호 ('-' 없이 입력)"
+            />
```
```/^(01[016789]{1})[0-9]{3,4}[0-9]{4}$/```
전화번호 regex rules를 추가해 유효성 검증 로직이 필요할 것 같아요

goodbyebada commented on src/utills/getPlatform.ts:
기기별 호환성을 고려하셨네요 👍

goodbyebada reviewed (approved):
작성하신 컴포넌트들이 Atomic 디자인 패턴을 잘 따르고 있어서, 재사용성과 유지보수성이 높은 것 같아요. 각 컴포넌트가 독립적으로 잘 설계되어 있어, 프로젝트 확장 시 많은 도움이 될 것 같아요.
고생하셨습니다!


- [x] 회원 프로세스가 사용자 측면에서 적합한가
- [x]  추가된 atoms 컴포넌트들이 재사용하기에 적합한가
- [x]  기존에 요소들과 중복되는 부분은 없는가

nyewon commented on src/components/atoms/Container.tsx:
Container나 Form를 따로 분리해서 다양한 상황에 적용할 수 있고, 컴포넌트를 재사용할 수 있다는 게 좋은 것 같아요!

nyewon commented on src/components/organisms/QRScanner.tsx:
```diff
@@ -0,0 +1,51 @@
+  highlightCodeOutline: true,
+};
+
+const QRScanner = ({ setValue }: QRScannerProps) => {
```
QrScanner 라이브러리를 처음 보는데 신기하네요! QR 코드 스캔 기능을 구현한 부분이 잘 구현 된 것 같습니다!
